runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java

```diff
@@ -80,5 +93,86 @@
    * PCollection PCollections} will be the {@link PCollectionNode PCollections} returned by {@link
    * #getOutputPCollections()}.
    */
-  PTransform toPTransform();
+  public final PTransform toPTransform() {
```

Comment:
Why not keep this as an interface and make this a default method?  Targeting Java 8, the static method below can also live in the interface.

Reply:
I want to force an overload of equals and hashCode, and got pushback from my compiler (I think). Certainly AutoValue clobbered me there.

Reply:
Since we no longer need `hashCode` or `equals`, I prefer an interface for aesthetic reasons. I'll leave the final judgement up to you between this and ensuring that `toPTransform` is final.

Reply:
Done.

To be honest, this has many of the same restrictions as any `XPayload` object - it's got a known format, but all translators using the URN must produce the same format of stage, otherwise either preconditions will fail or code will be buggy.

Documented that proto shape, and moved to an interface.